modules/search-pipeline-common/src/main/java/org/opensearch/search/pipeline/common/OversampleRequestProcessor.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.pipeline.common;
+
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.ingest.ConfigurationUtils;
+import org.opensearch.search.pipeline.AbstractProcessor;
+import org.opensearch.search.pipeline.Processor;
+import org.opensearch.search.pipeline.SearchRequestProcessor;
+import org.opensearch.search.pipeline.StatefulSearchRequestProcessor;
+import org.opensearch.search.pipeline.common.helpers.ContextUtils;
+
+import java.util.Map;
+
+import static org.opensearch.search.pipeline.common.helpers.ContextUtils.applyContextPrefix;
+
+/**
+ * Multiplies the "size" parameter on the {@link SearchRequest} by the given scaling factor, storing the original value
+ * in the request context as "original_size".
+ */
+public class OversampleRequestProcessor extends AbstractProcessor implements StatefulSearchRequestProcessor {
+
+    /**
+     * Key to reference this processor type from a search pipeline.
+     */
+    public static final String TYPE = "oversample";
+    static final String SAMPLE_FACTOR = "sample_factor";
+    static final String ORIGINAL_SIZE = "original_size";
+    private final double sampleFactor;
+    private final String contextPrefix;
+
+    private OversampleRequestProcessor(String tag, String description, boolean ignoreFailure, double sampleFactor, String contextPrefix) {
+        super(tag, description, ignoreFailure);
+        this.sampleFactor = sampleFactor;
+        this.contextPrefix = contextPrefix;
+    }
+
+    @Override
+    public SearchRequest processRequest(SearchRequest request, Map<String, Object> requestContext) {
+        if (request.source() != null) {
+            int originalSize = request.source().size();
+            requestContext.put(applyContextPrefix(contextPrefix, ORIGINAL_SIZE), originalSize);
+            int newSize = (int) Math.ceil(originalSize * sampleFactor);
```

Comment:
should we have some validation here so that size is not going above like 10k which is max size value for OpenSearch?

Reply:
That max is an index setting and we don't (easily) have access to index settings in search processors. 

(Technically, on startup, the module gets a reference to `SearchPipelineService`, which exposes a reference to `ClusterService`, which exposes `ClusterState`, which has `Metadata`, including `IndexMetadata`.)

I think I'm okay with the processor doing what it was asked, and letting it throw an exception when it gets to `DefaultSearchContext.preProcess`. What do you think?

Reply:
I think thats fine. As long as some will throw the exception and don't let the request process.

Resolving the comment.